src/network/mod.rs

```diff
@@ -2861,42 +2863,71 @@ impl NetworkingSystem {
         self.login_data = LoginData::new(
             login_server_login_success_packet.account_id,
             login_server_login_success_packet.login_id1,
+            login_server_login_success_packet.login_id2,
             login_server_login_success_packet.sex,
         )
         .into();
 
-        let character_server_information = login_server_login_success_packet
-            .character_server_information
-            .into_iter()
-            .next()
-            .ok_or("no character server available")?;
+        let character_server_information_list: Vec<CharacterServerInformation> =
+            login_server_login_success_packet.character_server_information.into();
+
+        if character_server_information_list.len() == 0 {
+            return Err("no character server available".to_string());
+        }
```

Comment:
I'm not so sure about this part, but mostly of that are working great.

Reply:
I updated the code slightly to clean up the `into` thing a bit but I think returning an `Err` here is actually perfect.

Side note: I would really recommend you to install `clippy`